src/Marten/Events/IEventStoreOptions.cs

```diff
@@ -56,6 +57,52 @@ public interface IEventStoreOptions
         /// <param name="eventTypeName">Mapped CLR event type</param>
         void MapEventType(Type eventType, string eventTypeName);
 
+
+        /// <summary>
+        /// Maps CLR event type as particular event type name allowing to provide custom transformation.
+        /// This is useful for event type migration.
+        /// See more in docs: https://martendb.io/events/versioning.html#event-type-name-migration
+        /// </summary>
+        /// <param name="eventTypeName">Event type name</param>
+        /// <param name="jsonTransformation">Event payload transformation</param>
+        /// <typeparam name="TEvent">Mapped CLR event type</typeparam>
+        IEventStoreOptions Upcast<TEvent>(
+            string eventTypeName,
```

Comment:
Given the type name is optional in the abstract class version, could it be made optional with the fluent syntax?

Reply:
Currently, it's not optional; it's only optional if you use upcasting by the specific CLR type. See: https://github.com/JasperFx/marten/pull/2314/files?diff=split&w=1#diff-aeac6bc19425c3715f9bf9d7e6436e1a6f7a8f51e8d8b14dff919691ae00fc79R39

I might try to provide overload with two event types that would not require passing that if that helps.

Reply:
I see. Yes, that would be useful 👍

Reply:
Added new methods reusing the slimmed upcasting via CLR types: https://github.com/JasperFx/marten/pull/2314/commits/08873ff478c8ab469521476f699f75422ad7928c. Thoughts? :)

Reply:
Looks good, thanks!

Reply:
I thank you! 😅 Good call 🙂